static/analysis.go

```diff
@@ -69,3 +80,63 @@ func analyzeIdentityCall(c *ssa.Call) configkit.Identity {
 
 	return ident
 }
+
+// analyzeHandler analyzes the application's handler from the given type to
+// deduce its handler configuration.
+//
+// It is assumed that the caller knows what type this handler is and passes
+// handler's type through ht parameter.
+func analyzeHandler(
+	prog *ssa.Program,
+	typ types.Type,
+	ht configkit.HandlerType,
+) configkit.Handler {
+	hdr := &entity.Handler{
+		HandlerTypeValue: ht,
+		TypeNameValue:    typ.String(),
+		MessageNamesValue: configkit.EntityMessageNames{
+			Produced: message.NameRoles{},
+			Consumed: message.NameRoles{},
+		},
+	}
+
+	pkg := getTypePkg(typ)
+	fn := prog.LookupMethod(typ, pkg, "Configure")
+
+	for _, c := range findConfigurerCalls(fn) {
+		args := c.Common().Args
+
+		switch c.Common().Method.Name() {
+		case "Identity":
+			hdr.IdentityValue = analyzeIdentityCall(c)
+		case "ConsumesCommandType":
+			hdr.MessageNamesValue.Produced.Add(
+				message.NameFromType(
+					args[0].(*ssa.MakeInterface).X.Type(),
+				),
+				message.CommandRole,
+			)
+		case "ProducesEventType":
+			hdr.MessageNamesValue.Consumed.Add(
+				message.NameFromType(
+					args[0].(*ssa.MakeInterface).X.Type(),
+				),
+				message.EventRole,
+			)
+		}
+	}
+
+	return hdr
+}
+
+// getTypePkg returns a package from the given type.
+func getTypePkg(typ types.Type) *types.Package {
+	switch t := typ.(type) {
+	case *types.Named:
+		return t.Obj().Pkg()
+	case *types.Pointer:
+		return t.Elem().(*types.Named).Obj().Pkg()
```

Comment:
If this isn't a named type it will panic with a type assertion failure rather than the meaningful panic message you wrote below.

```suggestion
		return pkgOfNamedType(t.Elem())
```